### PR TITLE
config,util: Add setting for TLS version

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Problem Summary:

This is a follow up for https://github.com/pingcap/tidb/pull/26907#issuecomment-893339820 

This makes it possible to configure the minimum version of the TLS protocol that the server supports. This makes it possible to disable older TLS versions that are less secure.

Note that the PCI-DSS standard doesn't allow the use of TLSv1.0 and recommends TLSv1.2 or newer.

See also [Deprecating TLSv1.0 and TLSv1.1](https://tools.ietf.org/id/draft-ietf-tls-oldversions-deprecate-02.html) from the IETF

### What is changed and how it works?

- The minimum version is read from the config.
- This sets the minimum version to TLSv1.1 by default. Before this TLSv1.0 was also allowed.
- It is still possible to set this to allow TLSv1.0 (e.g. to support MySQL 5.6 client libraries)

### Check List 

Tests 

- [x] Manual test (add detailed scripts or steps below)

Side effects

- [x] Breaking backward compatibility (disabling TLSv1.0 by default)

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
The `tls-version` setting as added to set the minimum version of the TLS protocol.
```

### CODE REVIEW

**Comment by dveeden** [comment]
MySQL has a list of supported TLS versions. This PR only sets the minimum version.
https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_tls_version

**Review by disksing — APPROVED** [review]

**Comment by disksing** [comment]
Please help to take a look. /cc @tiancaiamao @jackysp

File: util/misc.go [hunk 1]
```diff
@@ -466,6 +466,28 @@ func LoadTLSCertificates(ca, key, cert string, autoTLS bool) (tlsConfig *tls.Con
...
+		minTLSVersion = tls.VersionTLS13
+	case "":
+	default:
+		logutil.BgLogger().Warn(
+			"Invalid TLS version, using default instead",
+			zap.String("tls-version", tlsver),
+		)
+	}
+	if minTLSVersion < tls.VersionTLS12 {
+		logutil.BgLogger().Warn(
```
Comment (jackysp) [inline util/misc.go:487]: If the user doesn't set TLS config, will the server always prints this log?

Comment (dveeden) [reply util/misc.go:487]: Yes as the default is to set the minimum version to TLSv1.1 which give better compatibility but isn't recommended when it comes to security.

**Review by tiancaiamao — APPROVED** [review]

**Review by jackysp — APPROVED** [review]

**Pull request merged**